docs/developers_guide/FlinkComponents.md

```diff
@@ -1,22 +1,72 @@
+  or generic one [UniversalKafkaSourceFactory](https://github.com/TouK/nussknacker/blob/staging/utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/source/UniversalKafkaSourceFactory.scala).
+  Those Kafka sources also have a [context initializer](https://github.com/TouK/nussknacker/blob/staging/utils/kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/kafka/source/KafkaContextInitializer.scala)
+  defined that adds additional `#inputMeta` variable with Kafka record metadata like: partition, topic, offset, etc.
+- [GenericSourceWithCustomVariablesSample](https://github.com/TouK/nussknacker/blob/staging/engine/flink/management/dev-model/src/main/scala/pl/touk/nussknacker/engine/management/sample/source/GenericSourceWithCustomVariablesSample.scala)
+  a source that emits defined elements in a parameter but leverages [generic node transformation](Components.md#genericnodetransformation), context initializing, generating test data, typing.
 
 ## Sinks
 
-Sinks are defined using [FlinkSink](https://github.com/TouK/nussknacker/blob/staging/engine/flink/components-api/src/main/scala/pl/touk/nussknacker/engine/flink/api/process/FlinkSink.scala). Again, `BasicFlinkSink` is provided for simple cases. 
+Sinks are defined using [FlinkSink](https://github.com/TouK/nussknacker/blob/staging/engine/flink/components-api/src/main/scala/pl/touk/nussknacker/engine/flink/api/process/FlinkSink.scala).
+Again, `BasicFlinkSink` is provided for simple cases which requires implementing a Flink's `SinkFunction`
+and corresponding `FlinkSink` to register arbitrary `DataStreamSink`.
+
+Similarly, Flink provides [basic](https://ci.apache.org/projects/flink/flink-docs-master/docs/dev/datastream/overview/#data-sinks) sinks
+and [connectors](https://ci.apache.org/projects/flink/flink-docs-master/docs/connectors/datastream/overview) which can be used while implementing
+own Nussknacker sinks. And again, Nussknacker has built-in sinks for writing to Kafka registered in
+[`FlinkKafkaComponentProvider`](https://github.com/TouK/nussknacker/blob/staging/engine/flink/components/kafka/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/FlinkKafkaComponentProvider.scala).
+
+Examples:
+- TODO simple sink ???
+- [FlinkKafkaUniversalSink](https://github.com/TouK/nussknacker/blob/staging/engine/flink/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/sink/flink/FlinkKafkaUniversalSink.scala)
+  and its factory [UniversalKafkaSinkFactory](https://github.com/TouK/nussknacker/blob/staging/utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/sink/UniversalKafkaSinkFactory.scala)
 
 ## Custom transformers
```

Comment:
I consulted Google and it does not seem a Flink term. If it is commonly used in Nu code, I would define this term in this document.